File: ch01/python/1.5.4.py
import numpy as np

A = np.array([
    [1, 2],
    [3, 4]
])

print(A)
# [[1 2]
#  [3 4]]
print(A.shape) # (2, 2)

B = np.array([
    [3, 0],
    [0, 6]
])

print(A + B)
#[[ 4  2]
# [ 3 10]]
print(A * B)
#[[ 3  0]
# [ 0 24]]
print(A * 10)
# [[10 20]
#  [30 40]]

File: ch01/elixir/scripts/1.5.4.exs
a = Nx.tensor([
  [1, 2],
  [3, 4]
], type: :u64)

dbg(a)
dbg(Nx.shape(a))
dbg(Nx.type(a))

b = Nx.tensor([
  [3, 0],
  [0, 6]
], type: :u64)

dbg(b)
dbg(Nx.add(a, b))
dbg(Nx.multiply(a, b))

dbg(a)
dbg(Nx.multiply(a, 10))
